pkg/apis/catalog/v1alpha1/derivedcustomresource_types.go

```diff
@@ -38,6 +41,12 @@ type VersionExposeConfig struct {
 	// specifies the fields that should be present in the derived CRD.
 	// +kubebuilder:validation:MinItems=1
 	Fields []FieldPath `json:"fields"`
+
+	// Patch which will be applied for every derived CRD instance before
```

Comment:
now that I'm thinking...maybe we ought to first apply the patch (call it `defaults`, or `pre-set`) and then the user's tenantObj values. That way the user has the last say which values are going to be applied....actually that makes more sense

P.S.
I'd like baseObj/derivedObj notation versus tenantObj/ProviderObj. What do you think?

Reply:
What do you mean by user here? You mean tenant?

Reply:
Where do you see the `user` anymore?

Reply:
Just in your comment: https://github.com/kubermatic/kubecarrier/pull/312#discussion_r395230670, I am not sure if you meant provider or tenant